graphql/src/main/java/com/linecorp/armeria/server/graphql/DefaultGraphqlService.java

```diff
@@ -92,38 +130,62 @@ protected HttpResponse executeGraphql(ServiceRequestContext ctx, GraphqlRequest
         }
 
         final ExecutionInput executionInput =
-                builder.context(ctx)
-                       .graphQLContext(GraphqlServiceContexts.graphqlContext(ctx))
-                       .dataLoaderRegistry(dataLoaderRegistryFunction.apply(ctx))
-                       .build();
+            builder.context(ctx)
+                .graphQLContext(GraphqlServiceContexts.graphqlContext(ctx))
+                .dataLoaderRegistry(dataLoaderRegistryFunction.apply(ctx))
+                .build();
         return execute(ctx, executionInput, produceType);
     }
 
+    /**
+     * WebSocket operations are handled here.
+     */
+    @Override
+    public ExecutionResult executeGraphql(ExecutionInput.Builder builder) {
+        requireNonNull(dataLoaderRegistrySupplier, "dataLoaderRegistrySupplier");
+        final ExecutionInput executionInput = builder
+            .dataLoaderRegistry(dataLoaderRegistrySupplier.get())
+            .executionId(ExecutionId.generate())
+            .build();
+
+        return graphQL.execute(executionInput);
```

Comment:
This looks like a blocking call on the event loop.
What do you think of reusing the logic at L164 `(DefaultGraphqlService#execute(ServiceRequestContext ctx, ExecutionInput input, MediaType produceType))` to return a future instead?

Reply:
Still need to consume the future in `GraphqlWSSubProtocol`. That would still block the event loop. Or could that use the `BlockingTaskExecutor` to establish the subscriber?